flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -792,6 +792,12 @@ public void configure(ReadableConfig configuration, ClassLoader classLoader) {
 			});
 		config.configure(configuration, classLoader);
 		checkpointCfg.configure(configuration);
+		configuration.getOptional(ExecutionOptions.SORT_INPUTS).ifPresent(
```

Comment:
Do we want to expose these to the user? Even if it is a power user. I lean more towards these being set depending on the selected execution mode in the `StreamGraphGenerator` (see https://github.com/apache/flink/pull/13656). This will also remove the need of passing the configuration to the generator

Reply:
I am rather strong on having an option to disable these features. 

First of all I think it is good to have feature flags, secondly it makes it possible to e.g. use the `BATCH` execution mode (scheduling + shuffles) with `InputSelectable`.

Reply:
The way it is now though, the user has to explicitly state that he/she wants the BATCH state backend and timer service, right? Not the other way around. I would expect that even when we have the options available, when the `runtime-mode` is decided to be BATCH, the default behaviour would be to use the batch statebackend and timerservice.

Reply:
Ok now I saw that the defaults are for the Batch statebackend and the timer service to be activated.

Reply:
Yes, the options take effect only if the `RuntimeExecutionMode.BATCH` is enabled.